vlm/src/main/tut/RasterSource.md

```diff
@@ -0,0 +1,113 @@
+# RasterSource
+
+`RasterSource` interface is fundamental raster input interface in GeoTrellis.
+It abstracts over, file format, storage method and read implementation.
+
+This code is currently being developed under [geotrellis-contrib](https://github.com/geotrellis/geotrellis-contrib) project with plans of merging it with GeoTrellis 3.0 release.
+
+Current implementations exist for:
+
+- GeoTrellis GeoTiff Reader
+- GeoTrellis Indexed Layer
+- GDAL
+
+## Metadata Access
+
+Metadata will be read separately from pixels and user should expect it to be cached for repeated accessed through available fields.
+Additionally `RasterSource` should always be able to describe itself using a URI string.
```

Comment:
`MosaicRasterSource` is in conflict with this -- should this be qualified to non-collection `RasterSource`s, now that `MosaicRasterSource`s are planned?

Reply:
Thats a good point. I'm not sure what to do about it. `MosaicRasterSource` extending `RasterSource` is what makes it useful. Do you think it would be accurate to switch to `uri: List[String]` where the list order can be overloaded to mean z-order ?